source/posts/2020-09-10-cross-browser-testing-with-non-functional-tests.html.markdown.erb

```diff
@@ -0,0 +1,194 @@
+- [wikipedia](https://en.wikipedia.org/wiki/Non-functional_requirement).
+
+In my roles as a developer and tester, I see teams only using automated 
+UI tests, originally written to test the functionality, run across browsers to 
+ensure compatibility. Here, functional tests are used as a blanket to cover 
+the cross browser issues. 
+
+However, after working on [Gauge](https://gauge.org/) and [Taiko](https://docs.taiko.dev/), 
+I feel there’s a better and faster way to check cross browser issues than 
+relying only on functional tests. 
+
+In this post we look at ways to ensure cross browser compatibility without 
+relying on functional tests. Not all cross browser compatibility issues are 
+caught by functional tests. Here, I want to highlight how teams are using 
+alternatives to address them.
+
+## Deal with Non Functional Requirements earlier
+
+From the definition of BDD or Behaviour Driven Development, we know that,
+
+> The plan for implementing non-functional requirements is detailed in 
+the [system architecture](https://en.wikipedia.org/wiki/Systems_architecture), 
+because they are usually [architecturally significant requirements](https://en.wikipedia.org/wiki/Architecturally_significant_requirements). 
+- [wikipedia](https://en.wikipedia.org/wiki/Non-functional_requirement).
+
+Cross browser requirements have to be taken into account by developers 
+before the development begins as it’s architecturally significant.
+
+Most probably everyone uses some kind of framework which already checks 
+for compatibility. Javascript frameworks like [ReactJS](https://reactjs.org/), 
+[vue.js](https://vuejs.org/), [jquery](https://jquery.com/), and UI frameworks 
+like [bootstrap](https://getbootstrap.com/), [tailwind](https://tailwindcss.com/) 
+test for cross browser compatibility.
+
+Here are a few blogs that talk about the same
+
+* “React implements a totally different DOM system which is independent 
+of the browser. The implementation is done to make the application 
+cross-browser compatible.” - Cross-Browser Compatibility and React.js Web Apps
+* “Vue.js guide to ensure cross browser compatibility” - Browser Compatibility
+* “If you are using Twitter Bootstrap, and have been curious enough to 
+inspect the core bootstrap.css stylesheet, you will find approximately 
+6,000 lines of pure CSS code.  If you delve in deeper and look at the style 
+code in detail, you will find how they use multiple tricks and techniques to 
+make sure Bootstrap is the most cross browser CSS framework” 
+- [cross browser compatibility tips](https://www.monitis.com/blog/cross-browser-compatibility-tips-part-1/)
+
+It is better to invest in the right architecture for your application than 
+waiting for issues to be reported.
+
+## Write non functional tests for Cross browser compatibility
+
+To understand what we expect from a cross browser test, let’s analyse how 
+the browser is rendered. There are 3 main components to how browser 
+rendering works. 
```

Comment:
Change this whole paragraph to

There are three main components to how a browser renders a web page.